src/shutdown.cc

```diff
@@ -499,7 +535,16 @@ void do_system_shutdown(shutdown_type_t shutdown_type)
 #ifdef __NetBSD__
     reboot(reboot_type, NULL);
 #else
-    reboot(reboot_type);
+    if (reboot(reboot_type)) {
+        // we're in trouble now
+        sub_buf.append("reboot(2): ");
+        sub_buf.append(strerror(errno));
+        sub_buf.append(
+                "\nThis may happen if you try to kexec without loading an image first, or if\n"
+                "\nsomehow a reboot type unsupported by the kernel is attempted\n"
+        );
+        loop.poll();
```

Comment:
This should really be calling `loop.run()`, until `sub_buf` is drained (or perhaps forever in an infinite loop, to avoid a kernel panic).

Calling `loop.poll()` only checks if output is possible immediately, once. In the case the console is a serial tty for example, it might not be, and in that case, the important error message will never be shown before the kernel panics.